### PR TITLE
Docker: Separate ol-web1 and ol-web2 profiles with different hostnames

### PR DESCRIPTION
**Pull request by cclauss**
1. Create `web` and `web2` Docker nodes that have different `hostname` and `profiles` but share all other keys and values.
2. Add `test_docker_web_nodes()` to `test_docker_compose.py`
3. Move `tests/test_docker_compose.py` to `openlibrary/tests/test_docker_compose.py` to fix failing local test:
    * `docker compose run web pytest tests/test_docker_compose.py` --> `ModuleNotFoundError: 'openlibrary'`
    * `docker compose run web pytest openlibrary/tests/test_docker_compose.py` --> `4 passed in 0.46s`

### CODE REVIEW

**Comment by cclauss** [comment]
@cdrini Please let me know your thoughts on this PR.  It has been open for many weeks so we should decide if it is useful.

File: compose.production.yaml [hunk 1]
```diff
@@ -7,11 +7,12 @@
 
 version: "3.8"
 services:
-  web:
-    profiles: ["ol-web1", "ol-web2"]
+  # Use YAML anchor (&) and alias (*) to provide different profiles and hostnames
+  web:  &web_configuration
```
Comment (cdrini) [inline compose.production.yaml:11]: Note compose files inherit from each other (and this `web`  section inherits from `compose.yaml`). Does inheriting work correctly? So when we create web2 does web2 have all the inherited properties? For example, the `command` field is inherited.

Try adding `web2` to `compose.override.yaml`. That'll let you test this configuration out locally.

Comment (cclauss) [reply compose.production.yaml:11]: Does `test_docker_web_nodes()` not cover these conditions?

Comment (cdrini) [reply compose.production.yaml:11]: That tests just the yaml we need to confirm this actually runs correctly when using docker. The python test only checks the fields `profile` and `hostname`. We need to confirm that `command`, which is defined in compose.yaml, exists.

Either way, testing this locally first will give us a lot of clarity/confidence!

Comment (cclauss) [reply compose.production.yaml:11]: > The python test only checks the fields `profile` and `hostname`.

All keys and values are tested...
`assert web == web2  # All remaining keys and values should be the same.`

I take a look at `command`

**Pull request closed without merging**